File: FamilyTree/ClientApp/src/components/tree/model/Link.ts
import { EntityId } from "@reduxjs/toolkit";
import { getLinkId } from "../helpers/idHelpers";

export class Link {
  source: EntityId;
  target: EntityId;
  id: string;

  constructor(source: EntityId, target: EntityId) {
    this.id = getLinkId(source, target);
    this.source = source;
    this.target = target;
  }
}

File: FamilyTree/Models/Tree/Link.cs

namespace FamilyTree.Models
{
    public class Link
    {
        public Link(string source, string target)
        {
            Source = source;
            Target = target;
            Id = $"l{source}{target}";
        }
        //do serializacji
        public Link()
        {

        }
        
        public string Id { get; set; }
        public string Source { get; set; }
        public string Target { get; set; }
    }
}
